libs/ledger-live-common/src/exchange/swap/maybeTezosAccountUnrevealedAccount.ts

```diff
@@ -1,16 +1,25 @@
+import { Account, AccountLike } from "@ledgerhq/types-live";
 import { SwapTransactionType } from "./types";
+import { TezosAccount } from "../../families/tezos/types";
+
+// Type guard function to check if an account is a Tezos account
+function isTezosAccount(account?: Account | AccountLike): account is TezosAccount {
+  return (account as TezosAccount).tezosResources !== undefined;
+}
 
 export const maybeTezosAccountUnrevealedAccount = (
   swapTransaction: SwapTransactionType,
 ): Error | undefined => {
   if (
-    swapTransaction.swap?.rates.status == "success" &&
-    swapTransaction?.transaction?.family == "tezos" &&
-    swapTransaction?.transaction?.estimatedFees &&
-    swapTransaction?.transaction?.fees !== swapTransaction?.transaction?.estimatedFees
+    swapTransaction.swap?.rates.status === "success" &&
+    swapTransaction?.transaction?.family === "tezos"
```

Comment:
Since it seems quite redundant check the fees, I wonder if the Tezos network might have any issues that occasionally lead to forgetting to incorporate this check.

Reply:
Oh I'll add a comment, the fee check is to prevent rerender issue that crashed the app. 
Louis has done some improvement on this side I'll check if I can remove this hack

Reply:
This check was an hack before I realized there is a revealed field in the account. It was based on the assumptions that fees and estimated fees were different because only one of them include the reveal action (which adds fees)

this assumptions happened to be incorrect and we can check if an account is revealed in a simplier way